4-storage/rdb-loader/project/BuildSettings.scala

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2012-2017 Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+
+// SBT
+import sbt._
+import Keys._
+
+// sbt-assembly
+import sbtassembly._
+import sbtassembly.AssemblyKeys._
+
+/**
+ * Common settings-patterns for Snowplow apps and libraries.
+ * To enable any of these you need to explicitly add Settings value to build.sbt
+ */
+object BuildSettings {
+
+  /**
+   * Common set of useful and restrictive compiler flags
+   */
+  lazy val buildSettings = Seq(
+    scalacOptions ++= Seq(
+      "-deprecation",
+      "-encoding", "UTF-8",
+      "-feature",
+      "-unchecked",
+      "-Ywarn-unused-import",
+      "-Ywarn-nullary-unit",
+      "-Xfatal-warnings",
+      "-Xlint",
+      "-Yinline-warnings",
+      "-language:higherKinds",
+      "-Ypartial-unification",
+      "-Xfuture"),
+
+    scalacOptions in (Compile, console) := Seq(
+      "-deprecation",
+      "-encoding", "UTF-8"
+    ),
+
+    addCompilerPlugin("org.spire-math" % "kind-projector" % "0.9.4" cross CrossVersion.binary)
```

Comment:
Sorry, @BenFradet, missed these points.

No, this plugin is to avoid type-lambdas, it's not supported by any versions of Scala.